Content.Server/GameObjects/Components/Interactable/ToolComponent.cs

```diff
@@ -0,0 +1,180 @@
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float SpeedModifier
+        {
+            get => _speedModifier;
+            set => _speedModifier = value;
+        }
+
+        public string UseSound
+        {
+            get => _useSound;
+            set => _useSound = value;
+        }
+
+        public string UseSoundCollection
+        {
+            get => _useSoundCollection;
+            set => _useSoundCollection = value;
+        }
+
+        public void AddQuality(ToolQuality quality)
+        {
+            _qualities |= quality;
+            Dirty();
+        }
+
+        public void RemoveQuality(ToolQuality quality)
+        {
+            _qualities &= ~quality;
+            Dirty();
+        }
+
+        public bool HasQuality(ToolQuality quality)
+        {
+            return _qualities.HasFlag(quality);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _audioSystem = _entitySystemManager.GetEntitySystem<AudioSystem>();
+            _interactionSystem = _entitySystemManager.GetEntitySystem<InteractionSystem>();
+            Owner.TryGetComponent(out _spriteComponent);
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            if (serializer.Reading)
+            {
+                var qualities = serializer.ReadDataField("qualities", new List<ToolQuality>());
+                foreach (var quality in qualities)
+                {
+                    AddQuality(quality);
+                }
+            }
+            serializer.DataField(ref _speedModifier, "speed", 1);
+            serializer.DataField(ref _useSound, "useSound", string.Empty);
+            serializer.DataField(ref _useSoundCollection, "useSoundCollection", string.Empty);
+        }
+
+        public virtual bool UseTool(IEntity user, IEntity target, ToolQuality toolQualityNeeded)
+        {
+            PlayUseSound();
+
+            return ActionBlockerSystem.CanInteract(user) && HasQuality(toolQualityNeeded);
+        }
+
+        protected void PlaySoundCollection(string name, float volume=-5f)
+        {
+            var soundCollection = _prototypeManager.Index<SoundCollectionPrototype>(name);
+            var file = _robustRandom.Pick(soundCollection.PickFiles);
+            _entitySystemManager.GetEntitySystem<AudioSystem>()
+                .Play(file, Owner, AudioParams.Default.WithVolume(volume));
+        }
+
+        public void PlayUseSound()
+        {
+            if(string.IsNullOrEmpty(UseSoundCollection))
+                _audioSystem.Play(UseSound, Owner);
+            else
+                PlaySoundCollection(UseSoundCollection, 0f);
+        }
+
+        public void AfterAttack(AfterAttackEventArgs eventArgs)
+        {
+            if (Qualities != ToolQuality.Prying)
```

Comment:
I did not tested your branch, nor I am familiar with the AfterAttack system, but if you are assembling/disassembling something with the crowbar, wouldn't that still trigger this function and try to remove the tile?

Also, for readability, I would have create a function for prying tiles:
```c#
if (Qualities == ToolQuality.Prying)
    TryPryTile(eventArg);
}
```

Reply:
From the [component interfaces wiki page](https://hackmd.io/@ss14/component-interfaces):
```IAfterAttack
This interface gives components a behavior when clicking on another object and no interaction occurs, or when interacting with an entity out of range that has no behavior. Doesn’t pass what you clicked on as an argument.```

Also that's a good suggestion for readability.